pkg/types/validation/installconfig.go

```diff
@@ -70,35 +69,19 @@ func validateNetworking(n *types.Networking, fldPath *field.Path) field.ErrorLis
 	for i, cn := range n.ClusterNetworks {
 		allErrs = append(allErrs, validateClusterNetwork(&cn, fldPath.Child("clusterNetworks").Index(i), &n.ServiceCIDR.IPNet)...)
 	}
-	if n.PodCIDR != nil {
-		if err := validate.SubnetCIDR(&n.PodCIDR.IPNet); err != nil {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("podCIDR"), n.PodCIDR, err.Error()))
-		}
-		if validate.DoCIDRsOverlap(&n.ServiceCIDR.IPNet, &n.PodCIDR.IPNet) {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("podCIDR"), n.PodCIDR, "podCIDR must not overlap with serviceCIDR"))
-		}
-	}
-	if len(n.ClusterNetworks) == 0 && n.PodCIDR == nil {
-		allErrs = append(allErrs, field.Invalid(fldPath, n, "either clusterNetworks or podCIDR is required"))
-	}
-	if len(n.ClusterNetworks) != 0 && n.PodCIDR != nil {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("podCIDR"), n.PodCIDR, "cannot use podCIDR when clusterNetworks is used"))
+	if len(n.ClusterNetworks) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("clusterNetworks"), "cluster network required"))
 	}
 	return allErrs
 }
 
-func validateClusterNetwork(cn *netopv1.ClusterNetwork, fldPath *field.Path, serviceCIDR *net.IPNet) field.ErrorList {
+func validateClusterNetwork(cn *types.ClusterNetworkEntry, fldPath *field.Path, serviceCIDR *net.IPNet) field.ErrorList {
 	allErrs := field.ErrorList{}
-	_, cidr, err := net.ParseCIDR(cn.CIDR)
-	if err == nil {
-		if validate.DoCIDRsOverlap(cidr, serviceCIDR) {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("cidr"), cn.CIDR, "cluster network CIDR must not overlap with serviceCIDR"))
-		}
-		if ones, bits := cidr.Mask.Size(); cn.HostSubnetLength > uint32(bits-ones) {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("hostSubnetLength"), cn.HostSubnetLength, "cluster network host subnet length must not be greater than CIDR length"))
-		}
-	} else {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("cidr"), cn.CIDR, err.Error()))
+	if validate.DoCIDRsOverlap(&cn.CIDR.IPNet, serviceCIDR) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("cidr"), cn.CIDR.String(), "cluster network CIDR must not overlap with serviceCIDR"))
+	}
+	if ones, bits := cn.CIDR.Mask.Size(); cn.HostSubnetLength > uint32(bits-ones) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("hostPrefix"), cn.HostSubnetLength, "cluster network host prefix must not be larger than CIDR"))
```

Comment:
`hostPrefix` is not the correct json name for the field: it is `hostSubnetLength`.

Reply:
oops, fixed.